packages/libs/src/services/discoveryProvider/DiscoveryProvider.ts

```diff
@@ -1082,14 +1082,67 @@ export class DiscoveryProvider {
   }
 
   async relay(
-    data: DiscoveryRelayBody
-  ): Promise<{ receipt: TransactionReceipt } | null | undefined> {
-    const req = {
-      endpoint: 'relay',
+    body: DiscoveryRelayBody
+  ): Promise<{ receipt: TransactionReceipt } | undefined> {
+    const requestObj: Record<string, unknown> = {
+      endpoint: '/relay',
       method: 'post',
-      data
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      data: body
+    }
+    if (!this.discoveryProviderEndpoint || !this.discoveryNodeMiddleware) return
+
+    const axiosRequest = this._createDiscProvRequest(
+      requestObj as RequestParams,
+      this.discoveryProviderEndpoint
+    )
+
+    const { data, url = '', ...restRequest } = axiosRequest
+
+    const fetchRequestInit: RequestInit = {
+      body: data ? JSON.stringify(data) : data,
+      ...restRequest
+    }
+    let fetchParams = { url, init: fetchRequestInit }
+
+    fetchParams =
+      (await this.discoveryNodeMiddleware.pre?.({ fetch, ...fetchParams })) ??
+      fetchParams
+    let response: globalThis.Response | undefined
```

Comment:
perhaps there's a more useful way of duplicating most of this logic? I'd like to not change any of the read endpoints if possible

Reply:
the discovery _makeRequest will parse out the `.data` field in the response and assumes all calls going through it are a query, relay however is not and that's why it can't be used as it stands

Reply:
Hmmm... Does `_makeRequestInternal` not work?